src/server/input/input_probe.cpp

```diff
@@ -88,32 +126,49 @@ mir::UniqueModulePtr<mi::Platform> mi::probe_input_platforms(
             return Selection::persist;
         };
 
+    // First, check if any already-loaded graphics modules double as an input platform
```

Comment:
Shouldn't this go after the manually-selected ones instead?

Reply:
We first search to see if our graphics platform doubles as an input platform _and_ it is manually selected. If that's the case, we can just return the already loaded platform (as opposed to loading it again when we try and get it manually).

Otherwise, we check for a manually loaded platform. And, if not that, we get the best possible platform.

Reply:
In that case, the comment below is wrong?

>     // Otherwise, try and load a manually specified platform, as that takes precedence over the loaded platform

It doesn't take precedence, then? But shouldn't it?